interpreter/src/lib.rs

```diff
@@ -380,28 +382,39 @@ mod tests {
 
     #[test]
     fn test_thread_end() {
```

Comment:
For this test you might want to introduce a seeded RNG to make it deterministic. The rest of the code looks fine now.